features/checkout/preventing_adding_to_cart_product_or_variant_from_another_channel.feature

```diff
@@ -0,0 +1,27 @@
+@checkout
+Feature: Preventing adding to cart product or variant from another channel
+    In order to have correct products in cart when adding them
+    As a customer
+    I want to have the added products validated
+
+    Background:
+        Given the store operates on a channel named "France" with hostname "france"
+        And the store has a "Baquette" configurable product
+        And this product has "Small", "Medium" and "Large" variants
+        And the store has a "Pain" product
+        And the store operates on a channel named "Poland" with hostname "poland"
+        And I am a logged in customer
+
+    @api
+    Scenario: Preventing customer from adding simple product from another channel
+        Given I am browsing channel "Poland"
+        When I pick up my cart
+        And I try to add product "Pain" to the cart
+        Then I should be informed that product "Pain" is not available in current channel
```

Comment:
```suggestion
        Then I should be informed that product "Pain" is not available
```
information about different channels shouldn't be provided to customer
now if you will add to cart a non-existent product you will get 500, so or we should have the same result here or validation message like "there is są product 'Pain' in shop'"